### PR TITLE
usersession,tests: clean ups for userd/settings.go and move xdgopenproxy under usersession

### PR DESCRIPTION
**Pull request by pedronis**
See individual commits.

### CODE REVIEW

**Comment by pedronis** [comment]
This is a follow up to #7129.

File: usersession/userd/settings.go [hunk 1]
```diff
@@ -127,17 +152,17 @@ func safeSnapFromSender(s *Settings, sender dbus.Sender) (string, *dbus.Error) {
...
 }
 
-func desktopFileFromValue(s *Settings, command string, setting string, dotDesktopValue string, sender dbus.Sender) (string, *dbus.Error) {
+func desktopFileFromValue(s *Settings, command string, setspec *settingSpec, dotDesktopValue string, sender dbus.Sender) (string, *dbus.Error) {
 	snap, err := safeSnapFromSender(s, sender)
 	if err != nil {
 		return "", err
 	}
-	if err := settingWhitelisted(setting); err != nil {
+	if err := setspec.validate(); err != nil {
```
Comment (bboozzoo) [inline usersession/userd/settings.go:160]: Feels like the validate is not really related to converting dotDesktopValue to corresponding snap_app.desktop. Perhaps it bit should be done similarly to `Get()/GetSub()` in the caller.

Comment (pedronis) [reply usersession/userd/settings.go:160]: Maybe true, but a bit unrelated to the overall refactoring here. I personally don't have a strong opinion either way.

Comment (pedronis) [reply usersession/userd/settings.go:160]: if we rename it to the very long desktopFileFromValueForSetting is probably clearer and a reasonable factoring

**Pull request merged**